src/java/picard/analysis/replicates/CollectIndependentReplicateMetrics.java

```diff
@@ -0,0 +1,539 @@
+
+        final Histogram<Byte> umiEditDistanceInDiffBiDups = new Histogram<>("editDistance", "diffAllelesCount");
+        final Histogram<Byte> umiEditDistanceInSameBiDups = new Histogram<>("editDistance", "sameAllelesCount");
+        final Histogram<Byte> alleleBalanceCount = new Histogram<>("alleleBalance", "alleleBalanceCount");
+
+        // get the intervals that correspond to het sites in the VCF
+        final SortedMap<QueryInterval, List<Allele>> intervalAlleleMap = getQueryIntervalsMap(VCF);
+        final Iterator<QueryInterval> queryIntervalIterator = intervalAlleleMap.keySet().iterator();
+
+        log.info("Found " + intervalAlleleMap.size() + " heterozygous sites in VCF.");
+
+        // get an iterator to reads that overlap the heterozygous sites
+        final SamReader in = SamReaderFactory.makeDefault().open(INPUT);
+
+        log.info("Querying BAM for sites.");
+
+        final SAMRecordIterator samRecordIterator = in.query(intervalAlleleMap.keySet().toArray(new QueryInterval[intervalAlleleMap.size()]), false);
+        final List<SamRecordFilter> samFilters = CollectionUtil.makeList(
+                new AlignedFilter(true),
+                new CountingPairedFilter(),
+                new SecondaryOrSupplementaryFilter(),
+                new MappingQualityFilter(MINIMUM_MQ)
+        );
+
+        final FilteringSamIterator filteredSamRecordIterator = new FilteringSamIterator(samRecordIterator, new AggregateFilter(samFilters));
+        log.info("Queried BAM, getting duplicate sets.");
+
+        // get duplicate iterator from iterator above (this is where the magic happens!)
+        final DuplicateSetIterator duplicateSets = new DuplicateSetIterator(filteredSamRecordIterator, in.getFileHeader());
+
+        QueryInterval queryInterval = null;
+
+        log.info("Starting iteration on reads");
+        final ProgressLogger progress = new ProgressLogger(log, 10000000, "examined", "duplicate sets");
+
+        IndependentReplicateMetric locusData = new IndependentReplicateMetric();
+        boolean useLocus = true;
+        boolean newLocus = false;
+        int thirdAlleleInfos = 0;
+        Allele badAllele = null;
+        String offendingReadName = null;
+
+        set:
+        while (duplicateSets.hasNext()) {
+
+            final DuplicateSet set = duplicateSets.next();
+            final SAMRecord setRep = set.getRepresentative();
+            final QueryInterval setRepsInterval = queryIntervalFromSamRecord(setRep);
+
+            progress.record(setRep);
+            // if the current duplicate set no longer overlaps the query interval then null it (and handle it below)
+            // also move to the next variant if the previous variant is bad.
+            if (!useLocus || queryInterval != null && isCleanlyBefore(queryInterval, setRepsInterval)) {
+                if (!useLocus) {
+                    metric.nThreeAllelesSites++;
+                    if (++thirdAlleleInfos < 100) {
```

Comment:
just checking you saw that this is a log.debug...not actually emitted unless using VERBOSITY=DEBUG. The number will show up in the nThreeAllelesSites metric column...